generatedocs/data/actions/setscriptvar/examples.md

```diff
@@ -0,0 +1,136 @@
+    "settings": {
+        "name": "mylocalvar",
+        "type": "string",
+        "value": "My app Name with number for session {{ .Session }}"
+    }
+},
+{
+    "action": "openapp",
+    "settings": {
+        "appmode": "name",
+        "app": "{{ .ScriptVars.mylocalvar }}"
+    }
+}
+```
+
+Create a variable containing an integer and use it in a loop creating bookmarks numbered 1 to 5. Then in a different loop reset variable and delete the bookmarks.
+
+```json
+{
+    "action": "setscriptvar",
+    "settings": {
+        "name": "BookmarkCounter",
+        "type": "int",
+        "value": "0"
+    }
+},
+{
+    "action": "iterated",
+    "settings": {
+        "iterations": 5,
+        "actions": [
+            {
+                "action": "setscriptvar",
+                "settings": {
+                    "name": "BookmarkCounter",
+                    "type": "int",
+                    "value": "{{ add .ScriptVars.BookmarkCounter 1 }}"
+                }
+            },
+            {
+                "action": "createbookmark",
+                "settings": {
+                    "title": "Bookmark {{ .ScriptVars.BookmarkCounter }}",
+                    "description": "This bookmark contains some interesting selections"
+                }
+            }
+            
+        ]
+    }
+},
+{
+    "action": "setscriptvar",
+    "settings": {
+        "name": "BookmarkCounter",
+        "type": "int",
+        "value": "0"
+    }
+},
+{
+    "action": "iterated",
+    "disabled": false,
+    "settings": {
+        "iterations": 3,
+        "actions": [
+            {
+                "action": "setscriptvar",
+                "settings": {
+                    "name": "BookmarkCounter",
+                    "type": "int",
+                    "value": "{{ .ScriptVars.BookmarkCounter | add 1}}"
+                }
+            },
+            {
+                "action": "deletebookmark",
+                "settings": {
+                    "mode": "single",
+                    "title": "Bookmark {{ $element:=range.ScriptVars.BookmarkCounter }} {{ $element }}{{ end }}"
+                }
+            }
+        ]
+    }
+}
+```
+
+Combine two variables `MyArrayVar` and `BookmarkCounter` to create 3 bookmarks with the names `Bookmark one`, `Bookmark two` and `Bookmark three`.
+
+```json
+{
+    "action": "setscriptvar",
+    "settings": {
+        "name": "MyArrayVar",
+        "type": "array",
+        "value": "one,two,three,four,five",
+        "sep": ","
+    }           
+},
+{
+    "action": "setscriptvar",
+    "settings": {
+        "name": "BookmarkCounter",
+        "type": "int",
+        "value": "0"
+    }
+},
+{
+    "action": "iterated",
+    "disabled": false,
+    "settings": {
+        "iterations": 3,
+        "actions": [
+            {
+                "action": "createbookmark",
+                "settings": {
+                    "title": "Bookmark {{ index .ScriptVars.MyArrayVar .ScriptVars.BookmarkCounter }}",
```

Comment:
Pretty cool stuff!

Reply:
Indeed, good example!

Is it possible to do push and pop operations on array using setscriptvar and template stuff?

Reply:
Hmm no, since sescriptvar right recreates the variable, will think a bit about this

Reply:
you can append 

```json
{
    "action": "setscriptvar",
    "settings": {
        "name": "MyVarArray",
        "type": "array",
        "value": "val1,val2,val3",
        "sep": ","
    }
},
{
    "action": "setscriptvar",
    "settings": {
        "name": "MyVarArray",
        "type": "array",
        "value": "{{ range $i,$element:=(split .MyArray ",") }}{{if ne 0 $i}},{{ end }}{{$element}}{{ end }},val4",
        "sep": ","
    }
}
```
should produce an array with val1,val2,val3,val4

so possible but currently not nice
 {{ range $i,$element:=(split .MyArray ",") }}{{if ne 0 $i}},{{ end }}{{$element}}{{ end }},val4

Reply:
could make it nicer by adding a join function

Reply:
although pop can technically be done with `slice {{ slice .ScriptVars.MyArray 0 (add (len .ScriptVars.MyArray) -1) }}` we have two problems here, 
1. If the result is "", we will fail on the creation of the variable since we test towards ""
2. len .ScriptVars.MyArray will not evaluate to 0 on nil (as when done in code) and instead through an error

So will add the `join` function and example how to create an array adding to it when creating bookmarks, then iterate it to delete them

Reply:
@atluq , ok added the `join` function and made the more advanced scenario using it. Not exactly push/pop, but don't want to add functions that actually manipulates the value of the variables in the templates.

Reply:
Great, taking a look. Totally with you on no variable mutation in template functions.